packages/vc-verification/src/verifier/claim-issuer-verification.ts

```diff
@@ -43,30 +49,43 @@ export class ClaimIssuerVerification {
    * credentialResolver,
    * issuerResolver
    * );
-   * await issuerVerification.verifyIssuer('issuerDID', 'role');
+   * await issuerVerification.verifyIssuer('issuerDID', 'role', roleCredentialCache, roleDefCache);
    * ```
    * @param issuer DID of the issuer
    * @param role name of the role claim
+   * @param roleCredentialCache Cache to store role credentials
+   * @param roleDefCache Cache to store role definition
    * @returns VerificationResult
    */
   async verifyIssuer(
     issuer: string,
-    role: string
+    role: string,
+    roleCredentialCache: IRoleCredentialCache,
+    roleDefCache: IRoleDefinitionCache
   ): Promise<VerificationResult> {
     let currentIssuerDID = issuer;
     while (true) {
-      if (!(await this.verifyIssuerAuthority(role, currentIssuerDID))) {
+      if (
+        !(await this.verifyIssuerAuthority(
+          role,
+          currentIssuerDID,
+          roleCredentialCache,
+          roleDefCache
+        ))
+      ) {
         return verificationResult(false, ERRORS.IssuerNotAuthorized);
       }
       const roleIssuers = await this._issuerDefResolver.getIssuerDefinition(
-        role
+        role,
+        roleDefCache
       );
       if (roleIssuers && roleIssuers.did) {
         return verificationResult(true, '');
```

Comment:
Could you please remind why we don't check that issuer is in roleIssuers.did?

Reply:
So, the verification of issuer is already being done before this call. https://github.com/energywebfoundation/ew-credentials/blob/1747ee878ef05e3534dcc67717dcdd0b81515ad5/packages/vc-verification/src/verifier/claim-issuer-verification.ts#L69.

This check is to validate if the chain has ended or it it is a role based issuance then set the new issuers and role to be verified.